plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -91,6 +103,73 @@ data CostModelBase f =
 deriving via CustomJSON '[FieldLabelModifier (StripPrefix "param", CamelToSnake)] (CostModelBase CostingFun) instance ToJSON (CostModelBase CostingFun)
 deriving via CustomJSON '[FieldLabelModifier (StripPrefix "param", CamelToSnake)] (CostModelBase CostingFun) instance FromJSON (CostModelBase CostingFun)
 
+{- Note [Cost model parameters]
+We want to expose to the ledger some notion of the "cost model parameters". Intuitively, these should be all the numbers that appear in the cost model.
+
+However, there are quite a few quirks to deal with.
+
+1. CostModel is stuctured
+
+That is, it's a complex data structure and the numbers in question are often nested inside it.
+To deal with this quickly, we take the ugly approach of operating on the JSON representation of the model.
+We flatten this down into a simple key-value mapping (see 'flattenObject' and 'unflattenObject'), and then
+look only at the numbers.
+
+2. We use floats, not integers
+
+We'd really prefer to expose integers as our parameters - they're just better behaved, and really we'd like to use integers
+internally too for determinism reasons. So we pretend that we have integers by scaling up all our numbers by 1000 and taking
+the integral floor, at some loss of precision.
```

Comment:
I think that's what we'll do eventually in the real cost model, although we might have to use a bigger number than 1000.

For example the model for `eq_integer` currently looks like this:

```
  "cpu": {
            "arguments": {
                "slope": 9.10065769022107e-4,
                "intercept": 2.98807708812102
            },
            "type": "min_size"
        }
```
These numbers are for times in microseconds, so we're looking at `3+.00091*min(s1.s2)` microseconds to check equality of an s1-word integer and an s2-word integer. If we scale that up by 1000 and take the floor then it'll just  be a constant 3μs. I'm mulitplying by 10000 in `ExBudgeting.hs` but eventually that may have to change.  10^4 is very confusing, so maybe 10^6 would be more sensible.  If we're going to be using integers then we don't want the numbers to get _too_ big, but if the mulitpler's too small then we'll lose too much precision.